tracing/jaeger/http_client.go

```diff
@@ -55,21 +54,24 @@ func NewJaegerClient(client http.Client, baseURL *url.URL) (jaegerClient *Jaeger
 
 	// if cluster exists in tags, use it
 	query := models.TracingQuery{}
-	tags := map[string]string{
-		models.IstioClusterTag: conf.KubernetesConfig.ClusterName,
-	}
-	query.Tags = tags
 	query.End = time.Now()
 	query.Start = query.End.Add(-10 * time.Minute)
 	query.Limit = 100
 	prepareQuery(&urlTraces, jaegerService, query, false)
 	r, err := queryTracesHTTP(client, &urlTraces)
 
-	if r != nil && err == nil && len(r.Data) == 0 || err != nil {
-		log.Debugf("Error getting query for tracing. cluster tags will be disabled.")
+	if err != nil {
+		log.Debugf("[HTTP Jaeger] Error getting query for tracing. cluster tags will be disabled. %s", err.Error())
 		ignoreCluster = true
 	} else {
-		ignoreCluster = false
+		if r != nil && len(r.Data) == 0 {
+			log.Debugf("[HTTP Jaeger] Error getting query for tracing. cluster tags will be disabled. No data returned for: %s", urlTraces.String())
+			ignoreCluster = true
+		} else {
+			if includeJaegerClusterTag(r.Data) {
+				ignoreCluster = false
+			}
```

Comment:
Just so I understand, we make an initial query for traces to see if they have the cluster tag present and if they don't then for the rest of the queries we assume there's no cluster tag? Since Istio ~1.21 the cluster tag is now always present by default unless explicitly removed. I don't think we need `ignoreCluster` at all anymore?

Otherwise deciding whether or not to include the cluster tag this way may introduce problems. What if there cluster tag is present but there's just no tracing data yet?

Reply:
I think in that case the only risk would be if there are be no traces in Istio < 1.21 (That I don't think it is supported anymore), or if the cluster tag changes for some reason. In that case we could create a config option to have a workaround if this changes? Or being able to disable it if for some reason the tag is not present?

Reply:
For metrics we rely on all the istio standard labels to be present: https://istio.io/latest/docs/reference/config/metrics/#labels which includes cluster. If someone explicitly removes that then likely Kiali queries wouldn't work. Can we treat traces the same way? cluster is no different than the other default tags that we rely on like `app` or `service`. The only difference is that it was added later than the others.

Reply:
So, then we would use the cluster tag always? We were using it just in case of multi cluster. But I guess that would help when there are traces from other clusters, but maybe Kiali is configured for single cluster.

Reply:
Right I think Kiali has always used cluster in metrics queries: https://github.com/kiali/kiali/blob/54f4f79e1769bb2b1c650a3aef228df3dc019b54/graph/telemetry/istio/istio.go#L138

Reply:
@nrfox updated to use always the cluster tag. 
Also, I've modified the tests to validate there are not Traces where the workload/service/app from the specific cluster doesn't have traffic